hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -128,14 +128,14 @@ private HoodieFileGroupReader(HoodieReaderContext<T> readerContext, HoodieStorag
    */
   private void initRecordIterators() throws IOException {
     ClosableIterator<T> iter = makeBaseFileIterator();
-    if (inputSplit.getLogFiles().isEmpty()) {
+    Option<Pair<HoodieFileGroupRecordBuffer<T>, List<String>>> initializationResultOpt = recordBufferLoader.getRecordBuffer(
+        readerContext, storage, inputSplit, orderingFieldNames, metaClient, props, readerParameters, readStats, fileGroupUpdateCallback);
```

Comment:
I assume the intention is that the new "updates" can either come from the log files (from storage for read, compaction, etc.) or record iterator (from ingestion input for the merge handle).  Is there a plan to provide a new abstraction to represent this instead of `InputSplit` which currently represents base and log files only?

Reply:
This can be done separate, to make the code easier to follow.

Reply:
I agree w/ you take here. we can do it as a follow up patch.

Reply:
Fixed now.